godel_scan_analysis/src/scan_roughness_scoring.cpp

```diff
@@ -0,0 +1,120 @@
+#include "godel_scan_analysis/scan_roughness_scoring.h"
+
+#include "godel_scan_analysis/scan_algorithms.h"
+
+#include <ros/ros.h> // debug
+
+#include <math.h> // isfinite
+
+/*
+  I have a variety of ideas for making this code faster if it was ever needed:
+  1) Keep scan data as an array of x and array of y for better cache coherency on windowing
+  2) The windowing could be made to operate in O(n) - touching every point only once
+  3) Pre-calculate x values (which are known)
+  4) Pre-calculate colors
+  5) Perhaps return an index map like PCL does from the filter function so
+     we don't need to make a copy of the scan
+  6) Pre-allocate or use stack space (std::array?) for storage space
+*/
```

Comment:
If none of this is to be used as documentation then perhaps an issue with the __enhancement__ label should be created for it.